widgets/blueprintSources/src/BlueprintSources.tsx

```diff
@@ -1,15 +1,46 @@
-// @ts-nocheck File not migrated fully to TS
 import SplitterLayout from 'react-splitter-layout';
 import Actions from './actions';
 
-export default function BlueprintSources({ data, toolbox, widget }) {
+interface BlueprintTree {
+    children: any;
+    key: string;
+    title: string;
+    isDir: boolean;
+    timestamp: number;
+}
+
+interface BlueprintSourcesProps {
+    data: {
+        blueprintId: string;
+        blueprintTree: BlueprintTree;
+        importedBlueprintIds: string[];
+        importedBlueprintTrees: {
+            timestamp: number;
+            children: any;
+        };
+        yamlFileName: string;
+    };
+    toolbox: Stage.Types.Toolbox;
+    widget: Stage.Types.Widget;
+}
+
+interface ItemProps {
+    children: any;
+    key: string;
+    title: string;
+    isDir: boolean;
+}
+
+type HighlightTextProps = 'bash' | 'javascript' | 'json' | 'python' | 'yaml';
+
+export default function BlueprintSources({ data, toolbox, widget }: BlueprintSourcesProps) {
```

Comment:
Optional:

Seeing that we use `data.*` prefixes in many lines, we could maybe destructure it at the top of the function and use directly object properties?

```suggestion
export default function BlueprintSources({ data, toolbox, widget }: BlueprintSourcesProps) {
    const { blueprintId, blueprintTree, importedBlueprintIds, importedBlueprintTrees, yamlFileName } = data;
```

It's optional as I didn't check if we won't have shadowing issues.

Reply:
Hm... Maybe let's leave it then unchanged. Thanks for checking.